Analysis/OverlayRemoval/src/CheatedMCOverlayRemoval.cc

```diff
@@ -0,0 +1,370 @@
+      m_nMCPs = MCPs->getNumberOfElements();
+      int nKeptPFOs = 0;
+      int nRemovedPFOs = 0;
+      
+      //Create new collection without PFOs from MCPs that are overlay
+      for (int i=0; i<m_nMCPs; i++) {
+	MCParticle* mcp = (MCParticle*) MCPs->getElementAt(i);
+	if (mcp->isOverlay()) { 
+	  float weightPFOtoMCP = 0.0;
+	  float weightMCPtoPFO = 0.0;
+	  ReconstructedParticle* linkedPFO = getLinkedPFO( mcp , RecoMCParticleNav , MCParticleRecoNav , false , false , weightPFOtoMCP , weightMCPtoPFO );
+	  if ( linkedPFO == NULL ) {
+	    continue;
+	  }
+	  streamlog_out(DEBUG5) << "i = " << i << ": energy = " << linkedPFO->getEnergy () << std::endl;
+	  OutputOverlayCollection->addElement(linkedPFO);
+	  nRemovedPFOs++;
+	}
+      }
+
+      for (int i=0; i<m_nAllPFOs; i++) {
+	ReconstructedParticle* pfo = (ReconstructedParticle*) PFOs->getElementAt(i);
+	float weightPFOtoMCP = 0.0;
+	float weightMCPtoPFO = 0.0;
+	MCParticle* linkedMCP = getLinkedMCP( pfo, RecoMCParticleNav , MCParticleRecoNav , false , false , weightPFOtoMCP , weightMCPtoPFO );
+	if ( linkedMCP != NULL ) {
+	  if (linkedMCP->isOverlay()) streamlog_out(DEBUG5) << "i = " << i << ": energy = " << pfo->getEnergy () << ", gen status = " << linkedMCP->getGeneratorStatus() << std::endl;
+	  if (linkedMCP->isOverlay()) continue;
+	}
+	OutputPfoCollection->addElement(pfo);
+	nKeptPFOs++;
+      }
+
+      streamlog_out(DEBUG5) << "In event " << m_nEvt << ": Kept PFOs = " << nKeptPFOs << " VS removed number of PFOs = " << nRemovedPFOs << " VS total number of PFOs = " << m_nAllPFOs << std::endl;
+      pLCEvent->addCollection(OutputPfoCollection, _OutputPfoCollection.c_str() );
+      m_nEvt++ ;
+    }
+  catch(...)
+    {
+      streamlog_out(WARNING) << "Check : Input collections not found in event " << m_nEvt << std::endl;
+    }
+  streamlog_out(DEBUG) << "nevt = " << m_nEvt << std::endl;
+}
+
+
+EVENT::MCParticle* CheatedMCOverlayRemoval::getLinkedMCP( EVENT::ReconstructedParticle *recoParticle , LCRelationNavigator RecoMCParticleNav , LCRelationNavigator MCParticleRecoNav , bool getChargedMCP , bool getNeutralMCP , float &weightPFOtoMCP , float &weightMCPtoPFO )
+{
+  streamlog_out(DEBUG1) << "" << std::endl;
+  streamlog_out(DEBUG1) << "Look for MCP linked to Reconstructed Particle:" << std::endl;
+
+  MCParticle* linkedMCP{};
+  bool foundlinkedMCP = false;
+  double maxweightPFOtoMCP = 0.;
+  double maxweightMCPtoPFO = 0.;
+  int iPFOtoMCPmax = -1;
+  int iMCPtoPFOmax = -1;
+  const EVENT::LCObjectVec& MCPvec = RecoMCParticleNav.getRelatedToObjects( recoParticle );
+  const EVENT::FloatVec&  MCPweightvec = RecoMCParticleNav.getRelatedToWeights( recoParticle );
+  for ( unsigned int i_mcp = 0; i_mcp < MCPvec.size(); i_mcp++ )
+    {
+      double mcp_weight = 0.0;
+      double trackWeight = ( int( MCPweightvec.at( i_mcp ) ) % 10000 ) / 1000.0;
+      double clusterWeight = ( int( MCPweightvec.at( i_mcp ) ) / 10000 ) / 1000.0;
```

Comment:
```suggestion
      double trackWeight = MarlinUtil::getTrackWeight( MCPweightvec.at( i_mcp ) );
      double clusterWeight = MarlinUtil::getClusterWeight( MCPweightvec.at( i_mcp ) );
```
These were introduced very recently in https://github.com/iLCSoft/MarlinUtil/pull/36 but it would make sense to use them here since they are available now.

Reply:
Done

Reply:
Sorry, I forgot: You will need to `#include "marlinutil/MarlinUtil.h"` to make this work.

Reply:
I added the `#include ...` to the header but still get the same error: `error: ‘getTrackWeight’ is not a member of ‘MarlinUtil’`

Reply:
@tmadlener, As far as I can see this feature is not included in the v01-17 tag... So its only in the master currently...

Reply:
Yes, that is correct. We will make tags and probably another patch release after this one.

In principle adding it should make the CI pass. If you want to try and see how it works in CI you can use the `/cvmfs/clicdp.cern.ch/iLCSoft/lcg/100/nightly/x86_64-centos7-gcc10-opt/init_ilcsoft.sh` script for example. (should be reachable from the NAF).